dot/rpc/subscription/websocket_integration_test.go

```diff
@@ -18,13 +18,379 @@ import (
+						"params": ["6"],
+						"id": 7}`),
+			respErr: nil,
+			resp:    []byte(`{"jsonrpc":"2.0","result":false,"id":7}` + "\n"),
+		},
+		"state_unsubscribeStorage_request_#2_string_param_": {
+			request: []byte(`{
+    					"jsonrpc": "2.0",
+						"method": "state_unsubscribeStorage",
+						"params": ["4"],
+						"id": 7}`),
+			respErr: nil,
+			resp:    []byte(`{"jsonrpc":"2.0","result":true,"id":7}` + "\n"),
+		},
+		"state_unsubscribeStorage_request_#3_int_param": {
+			request: []byte(`{
+    					"jsonrpc": "2.0",
+						"method": "state_unsubscribeStorage",
+						"params": [6],
+						"id": 7}`),
+			respErr: nil,
+			resp:    []byte(`{"jsonrpc":"2.0","result":false,"id":7}` + "\n"),
+		},
+		"state_unsubscribeStorage_request_#4_int_param": {
+			request: []byte(`{
+    					"jsonrpc": "2.0",
+						"method": "state_unsubscribeStorage",
+						"params": [4],
+						"id": 7}`),
+			respErr: nil,
+			resp:    []byte(`{"jsonrpc":"2.0","result":true,"id":7}` + "\n"),
+		},
+		"chain_subscribeNewHeads_request": {
+			request: []byte(`{
+							"jsonrpc": "2.0",
+							"method": "chain_subscribeNewHeads",
+							"params": [],
+							"id": 8
+						}`),
+			respErr: nil,
+			resp:    []byte(`{"jsonrpc":"2.0","result":6,"id":8}` + "\n"),
+		},
+	}
+	for name, testCase := range testCases {
+		testCase := testCase
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+			ctrl := gomock.NewController(t)
+
+			wsconn, c, cancel := setupWSConn(t)
+			wsconn.Subscriptions = make(map[uint32]Listener)
+			wsconn.StorageAPI = modules.NewMockAnyStorageAPI(ctrl)
+			defer cancel()
+			go wsconn.HandleConn()
+			time.Sleep(time.Second * 2)
+
+			c.WriteMessage(websocket.TextMessage, testCase.request)
+
+			_, msg, err := c.ReadMessage()
+			require.NoError(t, err)
+			require.Equal(t, testCase.resp, msg)
+		})
+	}
+
+}
+
+func TestWSConn_InitBlockListener(t *testing.T) {
+	t.Parallel()
+	testCases := map[string]struct {
+		setBlocAPI bool
+		respErr    error
+		resp       []byte
+	}{
+		"blockAPI_not_set": {
+			setBlocAPI: false,
+			resp:       []byte(`{"jsonrpc":"2.0","error":{"code":null,"message":"error BlockAPI not set"},"id":1}` + "\n"),
+			respErr:    errors.New("error BlockAPI not set"),
+		},
+		"blockAPI_set": {
+			setBlocAPI: true,
+			resp:       []byte(`{"jsonrpc":"2.0","result":5,"id":1}` + "\n"),
+			respErr:    nil,
+		},
+	}
+
+	for name, testCase := range testCases {
+		testCase := testCase
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+			ctrl := gomock.NewController(t)
+
+			wsconn, c, cancel := setupWSConn(t)
+			wsconn.Subscriptions = make(map[uint32]Listener)
+			wsconn.StorageAPI = modules.NewMockAnyStorageAPI(ctrl)
+			defer cancel()
+			go wsconn.HandleConn()
+			time.Sleep(time.Second * 2)
+
+			if testCase.setBlocAPI {
+				wsconn.BlockAPI = modules.NewMockAnyBlockAPI(ctrl)
+			}
+
+			res, err := wsconn.initBlockListener(1, nil)
+			require.EqualError(t, err, testCase.respErr.Error())
```

Comment:
```suggestion
			if testCase.respErr != nil {
				require.EqualError(t, err, testCase.respErr.Error())
			} else {
				require.NoError(t, err)
			}
```

this would fix nil pointer dereference